can/message.py

```diff
@@ -101,68 +163,174 @@ def __str__(self):
             except UnicodeError:
                 pass
 
+        if self.channel is not None:
+            field_strings.append("Channel: {}".format(self.channel))
+
         return "    ".join(field_strings).strip()
 
     def __len__(self):
         return len(self.data)
 
     def __bool__(self):
+        # For Python 3
         return True
 
     def __nonzero__(self):
+        # For Python 2
         return self.__bool__()
 
     def __repr__(self):
-        data = ["{:#02x}".format(byte) for byte in self.data]
         args = ["timestamp={}".format(self.timestamp),
-                "is_remote_frame={}".format(self.is_remote_frame),
-                "extended_id={}".format(self.id_type),
-                "is_error_frame={}".format(self.is_error_frame),
                 "arbitration_id={:#x}".format(self.arbitration_id),
-                "dlc={}".format(self.dlc),
-                "data=[{}]".format(", ".join(data))]
+                "extended_id={}".format(self.is_extended_id)]
+
+        if self.is_remote_frame:
+            args.append("is_remote_frame={}".format(self.is_remote_frame))
+
+        if self.is_error_frame:
+            args.append("is_error_frame={}".format(self.is_error_frame))
+
         if self.channel is not None:
-            args.append("channel={!r}".format(self.channel))
+            args.append("channel={!r}".format(self.channel))                
+
+        data = ["{:#02x}".format(byte) for byte in self.data]
+        args += ["dlc={}".format(self.dlc),
+                "data=[{}]".format(", ".join(data))]
+
         if self.is_fd:
             args.append("is_fd=True")
             args.append("bitrate_switch={}".format(self.bitrate_switch))
             args.append("error_state_indicator={}".format(self.error_state_indicator))
+
         return "can.Message({})".format(", ".join(args))
 
-    def __eq__(self, other):
-        if isinstance(other, self.__class__):
-            return (
-                self.arbitration_id == other.arbitration_id and
-                #self.timestamp == other.timestamp and # allow the timestamp to differ
-                self.id_type == other.id_type and
-                self.dlc == other.dlc and
-                self.data == other.data and
-                self.is_remote_frame == other.is_remote_frame and
-                self.is_error_frame == other.is_error_frame and
-                self.is_fd == other.is_fd and
-                self.bitrate_switch == other.bitrate_switch
-            )
-        else:
-            return NotImplemented
-
-    def __ne__(self, other):
-        if isinstance(other, self.__class__):
-            return not self.__eq__(other)
-        else:
-            return NotImplemented
+    # Comparing messages by something other than identity (the default) was
+    # discussed in and removed as part of PR #413.
 
     def __hash__(self):
         return hash((
+            self.timestamp,
```

Comment:
In general floats should not be used for hashing, just like one should not compare floats for equality. Do we need to implement the hash method if all messages that are not exactly the same object are considered unequal?

Reply:
As far as I know implementing our own hashing won't give us anything if we are using the default `id` based equality comparison?

Reply:
@hardbyte Yeh, you are right! We should not override `__hash__()`.